gammapy/maps/wcsnd.py

```diff
@@ -669,24 +669,47 @@ def cutout(self, position, width, mode="trim"):
         cutout : `~gammapy.maps.WcsNDMap`
             Cutout map
         """
-        width = _check_width(width)
-        idx = (0,) * len(self.geom.axes)
-        c2d = Cutout2D(
-            data=self.data[idx],
-            wcs=self.geom.wcs,
-            position=position,
-            # Cutout2D takes size with order (lat, lon)
-            size=width[::-1] * u.deg,
-            mode=mode,
-        )
+        geom_cutout = self.geom.cutout(position=position, width=width, mode=mode)
 
-        # Create the slices with the non-spatial axis
-        cutout_slices = Ellipsis, c2d.slices_original[0], c2d.slices_original[1]
+        slices = geom_cutout.cutout_info["parent-slices"]
+        cutout_slices = Ellipsis, slices[0], slices[1]
 
-        geom = WcsGeom(c2d.wcs, c2d.shape[::-1], axes=self.geom.axes)
         data = self.data[cutout_slices]
 
-        return self._init_copy(geom=geom, data=data)
+        return self._init_copy(geom=geom_cutout, data=data)
+
+    def stack(self, other, weights=None, check=True):
+        """Stack cutout into larger map.
+
+        Parameters
+        ----------
+        other : `WcsNDMap`
+            Other map to stack
+        weights : `~numpy.ndarray`
+            Other map to be used as weights.
+        check : bool
```

Comment:
Is the `check` argument really needed? Why not always check if the géométrie is really a cutout from the other?

Reply:
Currently there is the problem, that we don't create necessarily all the cutouts from the same  parent geometry. The situation will improve a bit after #2450. Once #2450 is merged I'll rebase this PR and re-address you comment.

Reply:
I removed the `check` argument again...